### PR TITLE
Implement `blob_sidecar` event

### PR DESCRIPTION
**Pull request by tbenr**
Implements `blob_sidecar` SSE event.

Note that this diverges from block implementation: to capture locally produced blobs as well as the one arriving via gossip, I intercept blobs at `BlobSidecarPool` instead of `BlobSidecarManager`

Partially implements #7513

## Documentation

- [x] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [x] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: ethereum/statetransition/src/test/java/tech/pegasys/teku/statetransition/util/BlobSidecarPoolImplTest.java
Comment (lucassaldanha) [inline ethereum/statetransition/src/test/java/tech/pegasys/teku/statetransition/util/BlobSidecarPoolImplTest.java:1]: Do we need a test for only calling `newBlobSidecarSubscribers.deliver()` when the block is new?

File: data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/events/EventSubscriptionManager.java [hunk 1]
```diff
@@ -52,6 +54,7 @@
 public class EventSubscriptionManager implements ChainHeadChannel, FinalizedCheckpointChannel {
   private static final Logger LOG = LogManager.getLogger();
 
+  private Spec spec;
```
Comment (lucassaldanha) [inline data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/events/EventSubscriptionManager.java]: NIT: can be final

**Review by lucassaldanha — APPROVED** [review]

File: ethereum/statetransition/src/test/java/tech/pegasys/teku/statetransition/util/BlobSidecarPoolImplTest.java
Comment (tbenr) [reply ethereum/statetransition/src/test/java/tech/pegasys/teku/statetransition/util/BlobSidecarPoolImplTest.java:1]: Not sure what you mean here. I added a test `onNewBlobSidecar_shouldIgnoreDuplicates`. Does it cover, in conjunction with `onNewBlobSidecar_shouldIgnoreBlobsForAlreadyImportedBlocks`, your doubts?

**Comment by tbenr** [comment]
merging this 'cose it could be useful for devnet-9. Happy to update tests in a later PR if need.

**Pull request merged**